src/eth/storage/inmemory/inmemory_permanent.rs

```diff
@@ -479,11 +480,15 @@ impl InMemoryPermanentAccount {
 
     /// Converts itself to an account at a point-in-time.
     pub fn to_account(&self, point_in_time: &StoragePointInTime) -> Account {
+        let bytecode = self.bytecode.get_at_point(point_in_time).unwrap_or_default();
+        let code_hash = CodeHash::from_bytecode(bytecode.clone());
```

Comment:
It seems we are calculating the hash every time the conversion happens.

We should persist the code hash to the storage once and just read it when retrieving information